### DISCUSSION TITLE
Needing help : how to exclude Guest account from people search

### DISCUSSION THREAD

**Discussion by jeffangama** [microsoft-search/pnp-modern-search#1935] [Q&A] [answered]
Our Azure AD is with a lot of stuff. we managed to exclude a lot of technical account using Title<>"admin*"

I have one requirement left : **How to exclude Guest account from people search**
Because at the moment they show as User Icon : 

![image](https://user-images.githubusercontent.com/6776269/157646662-0c80af6d-2dc9-470d-acaf-96087f050384.png)

Again : thank you for the great work on the product

**Comment by wobba** [comment, marked as answer]
Take a look at https://www.techmikael.com/2018/01/excluding-sharepoint-online-service.html, and you probably want to add something like `-accountname:"#ext#"`.

**Reply by jeffangama** [reply]
Thank you, I tried putting this in Query, it return nothing

My current query is 

`{searchTerms} (Title<>"ADM*")`

How shall i edit the query to exclude any email containing #ext# ?
In the debug view, i see userName or WorkEmail, but i can't use it in the query such as
`{searchTerms} (Title<>"ADM*" and userName<>"*ext*")`

Thank you again,

**Reply by wobba** [reply]
You should read up on KQL over at https://docs.microsoft.com/en-us/sharepoint/dev/general-development/keyword-query-language-kql-syntax-reference#kql-operators-for-complex-queries

But the query will be something like:
`{searchTerms} (-Title:"ADM*" -userName:"#ext#")`

I like use `-` instead of `<>`, but that's semantics. Operators also has to be uppercase if you use them, `AND` not `and`, and `AND` is explicit between different properties. And pre-fix wildcard is not supported. Also see my post https://www.techmikael.com/2014/03/s15e01-kql-basics.html

**Reply by jeffangama** [reply]
Hi, thank you again. I can't get this username filtering working

my query : 
{searchTerms} (-UserName:"*#ext#*")
I also tried
{searchTerms} (-UserName:"#ext#")
{searchTerms} (-userName:"#ext#")
{searchTerms} (-UserName:"#EXT#")
{searchTerms} (-UserName:"*#EXT#*")

It retrieve 0 acount, when I expect to retrieve any account EXCEPT accounts emails containing "#ext#"
I use the LocalPeopleResults Scope

In the debug Ive 

```
"Title": "Jeff ANGAMA",
"UserName": "jeff.an_outlook.fr#EXT#@xxx.onmicrosoft.com"
```

Thank you

**Reply by wobba** [reply]
You should be using `-AccountName:"#ext#"`. I just copied what you wrote without checking 🙂

**Reply by jeffangama** [reply]
{searchTerms} (-AccountName:"#ext#")
{searchTerms} (-AccountName:"#EXT#")
{searchTerms} (-AccountName:"*#ext#*")

I get no results when using any of those (I shall get one result)

**Reply by wobba** [reply]
Hard to say what the problem is as you haven't shared your full setup. Works like a charm for me ;) You can test with the SP query tool as well and see if you ser any change.

**Reply by jeffangama** [reply]
- I use V4 pnp webpart. I added a search web part
- I added a result web part
- In the result web part i chose the LocalPeopleResults as source
- The query is : {searchTerms} (-AccountName:"#ext#")
- And I connected the two webparts
- Here are my settings as XML if you can replicate my issue on your side maybe ? https://1drv.ms/u/s!As1Sdx69xVW4gpRFQ1ZAuL0DiBXM6g?e=wGsKVu

**Reply by wobba** [reply]
Not sure what happens as it works fine in two of my tenants.

**Reply by jeffangama** [reply]
I workaround the issue by filtering out email that are not our domain

{searchTerms} AND (WorkEmail:"*xxx.fr" OR WorkEmail:"*yyy.mc")

Thank you !

**Reply by adumith** [reply]
Hi guys,

I have a similar situation. I am starting to work with this WebPart and I would like to know how to exclude disabled accounts, which properties I must use to obtain data such as telephone, department, location and IM ID.

Thank in advance
